src/util/map.test.js

```diff
@@ -76,15 +76,6 @@ describe("util/map", () => {
       const o: {+[string]: number} = {a: 1};
       const _: Map<string, number> = MapUtil.fromObject(o);
     });
-    it("statically rejects a map with keys not a subtype of string", () => {
-      const input: {[number]: string} = {};
-      input[12] = "not okay";
-      input[13] = "also not okay";
-      // $FlowExpectedError[incompatible-call]
-      MapUtil.fromObject(input);
-      // If that were valid, then `(result.keys(): Iterator<number>)`
-      // would contain the strings "12" and "13".
-    });
```

Comment:
As mentioned above I removed this because of the change above, which isn't good. I'd like to put this back in, but I'm not sure how to statically enforce it

Reply:
update: This has been re-added after pairing with @wchargin